src/Agent.Plugins/PipelineCache/Telemetry/PipelineCacheActionRecord.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Agent.Sdk;
+using Agent.Plugins.PipelineArtifact.Telemetry;
+using Microsoft.VisualStudio.Services.Content.Common.Telemetry;
+using Microsoft.VisualStudio.Services.BlobStore.WebApi;
+using Microsoft.VisualStudio.Services.BlobStore.Common.Telemetry;
+using Microsoft.VisualStudio.Services.PipelineCache.WebApi;
+using Microsoft.VisualStudio.Services.PipelineCache.WebApi.Telemetry;
+
+namespace Agent.Plugins.PipelineCache.Telemetry
+{
+    /// <summary>
+    /// Generic telemetry record for use with Pipeline Caching events.
+    /// </summary>
+    public class PipelineCacheActionRecord : PipelineCacheTelemetryRecord
+    {
+        public static long FileCount { get; private set; }
+
+        public PipelineCacheActionRecord(TelemetryInformationLevel level, Uri baseAddress, string eventNamePrefix, string eventNameSuffix, AgentTaskPluginExecutionContext context, uint attemptNumber = 1)
+            : base(
+            level: level, 
+            baseAddress: baseAddress, 
+            eventNamePrefix: eventNamePrefix, 
+            eventNameSuffix: eventNameSuffix, 
+            planId: Guid.Parse(context.Variables["system.planId"].Value), 
```

Comment:
>Guid.Parse(context.Variables["system.planId"].Value) [](start = 20, length = 52)

handle invalid guid's passed in

Reply:
These are set the system and can't be overridden by the user so they should always be safe. If these values are malformed then something is seriously wrong. Stored as guids on [TaskActivityData](https://dev.azure.com/mseng/AzureDevOps/_git/AzureDevOps?path=%2FReleaseManagement%2FService%2FReleaseManagement2%2FData%2FModel%2FTaskActivityData.cs&version=GBmaster) and are some of the [WellKnownDistributedTaskVariables](https://dev.azure.com/mseng/AzureDevOps/_git/AzureDevOps?path=%2FDistributedTask%2FClient%2FWebApi%2FWebApi%2FWellKnownDistributedTaskVariables.cs&version=GBmaster)